lib/membrane_opus/encoder.ex

```diff
@@ -132,18 +133,48 @@ defmodule Membrane.Opus.Encoder do
   end
 
   @impl true
-  def handle_buffer(:input, %Buffer{payload: data}, _ctx, state) do
-    case encode_buffer(state.queue <> data, state, frame_size_in_bytes(state)) do
-      {:ok, {[], rest}} ->
+  def handle_buffer(:input, %Buffer{payload: data, pts: input_pts}, _ctx, state) do
+    prepared_state =
+      if state.queue == <<>> do
+        %{state | pts_current: input_pts}
+      else
+        state
+      end
+
+    check_pts_integrity_flag =
+      if state.queue != <<>> do
+        true
+      else
+        false
+      end
+
+    case encode_buffer(
+           state.queue <> data,
+           prepared_state,
+           frame_size_in_bytes(state)
+         ) do
+      {:ok, [], state} ->
         # nothing was encoded
-        {[], %{state | queue: rest}}
+        {[], state}
 
-      {:ok, {encoded_buffers, rest}} ->
+      {:ok, encoded_buffers, state} ->
         # something was encoded
-        {[buffer: {:output, encoded_buffers}], %{state | queue: rest}}
+        check_pts_integrity(check_pts_integrity_flag, List.first(encoded_buffers), input_pts)
+        {[buffer: {:output, encoded_buffers}], state}
     end
   end
 
+  defp check_pts_integrity(true = _flag, %Buffer{pts: pts}, input_pts) do
+    if pts != input_pts do
+      raise """
+      PTS values are not continuous
+      """
+    end
+  end
+
+  defp check_pts_integrity(false = _flag, %Buffer{pts: _pts}, _input_pts) do
+  end
+
```

Comment:
In the another comment I have written that is should be deleted, but FYI if you want to write function that do some checks and raises if any of them is not met and does nothing otherwise,
1) If all checks pass, function should return `:ok` instead of `nil`
2) name of the function should match the pattern `ensure_<sth>!`